src/ObjectUtils/CoffFile.cpp

```diff
@@ -417,49 +417,130 @@ orbit_object_utils::CoffFileImpl::GetPrimaryRuntimeFunction(
 }
 
 ErrorMessageOr<std::vector<CoffFileImpl::UnwindRange>> CoffFileImpl::GetUnwindRanges() {
+  static constexpr std::string_view kErrorMessagePrefix = "Unable to load unwind info ranges: ";
   auto runtime_functions_or_error = GetRuntimeFunctions();
   if (runtime_functions_or_error.has_error()) {
-    return ErrorMessage{absl::StrFormat("Unable to load unwind info ranges: %s",
-                                        runtime_functions_or_error.error().message())};
+    return ErrorMessage{
+        absl::StrCat(kErrorMessagePrefix, runtime_functions_or_error.error().message())};
   }
   const llvm::ArrayRef<llvm::Win64EH::RuntimeFunction>& runtime_functions =
       runtime_functions_or_error.value();
 
+  // We have the RUNTIME_FUNCTIONs, but chained UNWIND_INFO means that a single function can be
+  // formed by multiple RUNTIME_FUNCTIONs, so we have to merge some. We use the following rule: we
+  // merge two RUNTIME_FUNCTIONs if:
+  // - they are adjacent (the end address of the first equals the start address of the second); and
+  // - they have the same primary RUNTIME_FUNCTION (where a RUNTIME_FUNCTION that doesn't have
+  //   chained UNWIND_INFO is its own primary), as this corresponds to the prologue of the function.
+  //
+  // Note that this rule doesn't allow for "holes": if two consecutive RUNTIME_FUNCTIONs are not
+  // perfectly adjacent, we will consider them two separate functions, even if they have the same
+  // primary RUNTIME_FUNCTION. However, we never observed this case.
+  //
+  // And note that we don't require the primary RUNTIME_FUNCTION to actually be part of the current
+  // set of RUNTIME_FUNCTIONs we are merging. We observed this as a very rare but valid case, on
+  // which more information can be found below.
   std::vector<UnwindRange> unwind_ranges;
-  // For a function, the "primary" RUNTIME_FUNCTION and the ones with UNW_FLAG_CHAININFO set (if
-  // any) form a tree rooted on the "primary" RUNTIME_FUNCTION. We merge all such RUNTIME_FUNCTIONs
-  // from the same tree into a single UnwindRange.
+  uint64_t last_primary_address{};
+  uint64_t largest_primary_address{};
+  uint64_t num_chained_not_with_largest_primary = 0;
```

Comment:
Reading `num_chained_not_with_largest_primary`, I don't really have an idea what this is supposed to be. Can you think of a better name or add a comment?

Reply:
I'm relying on the comment in `if (primary_address < largest_primary_address) {`, which I extended slightly. I would avoid adding yet another one on this matter. However, I renamed to `number_of_runtime_functions_whose_primary_is_not_the_latest_primary`, I hope it's good enough, I doubt I can do better without going over 100 characters :D

Reply:
Sounds good, it's long, but it's clear